frontend/src/components/Forms/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from "react";
+import { SubmitHandler, useForm, FormProvider } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import { Link as ReactLink } from "@mui/material";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import { useNavigate, useLocation } from "react-router-dom";
+
+import googleLogo from "../../../assets/googleLogo.svg";
+import githubLogo from "../../../assets/githubLogo.svg";
+import facebookLogo from "../../../assets/facebookLogo.svg";
+import twitterLogo from "../../../assets/twitterLogo.svg";
+import TextFieldInput from "../../Fields/FormInput/TextFieldInput";
+import CheckboxInput from "../../Fields/FormInput/CheckboxInput";
+import StatusCode from "../../../helpers/StatusCode";
+import Constants from "../../../helpers/Constants";
+import { UserType } from "../../../helpers/CustomTypes";
+import useFetchPlus from "../../../hooks/useFetchPlus";
+
+const validationSchema = z.object({
+  email: z
+    .string()
+    .min(1, { message: "Email is required" })
+    .email({ message: "Please enter a valid email address" }),
+  password: z
+    .string()
+    .min(1, "Password is required")
+    .min(8, "Password must be more than 8 characters")
+    .max(32, "Password must be less than 32 characters"),
+  remember: z.boolean(),
+});
+
+type ValidationSchema = z.infer<typeof validationSchema>;
+
+function LoginForm() {
+  const fetchPlus = useFetchPlus();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const from = location.state?.from?.pathname || "/";
+
+  const defaultValues: ValidationSchema = {
+    email: "",
+    password: "",
+    remember: false,
+  };
+
+  const methods = useForm<ValidationSchema>({
+    resolver: zodResolver(validationSchema),
+    defaultValues,
+  });
+
+  const onSubmitHandler: SubmitHandler<ValidationSchema> = async (
+    data: ValidationSchema
+  ) => {
+    fetchPlus(`${Constants.BASE_URL}/auth/login`, {
+      body: JSON.stringify({
+        email: data.email,
+        password: data.password,
+      }),
+    })
+      .then((response: Response) => {
```

Comment:
Why `.then` and not just `await` since this is already marked as an async method?

Reply:
I forgot to delete the `async` mark after some testing. Fixed that.

Reply:
As we discussed, I have changed the `.then` to `await`.